src/prefect/orion/orchestration/rules.py

```diff
@@ -263,7 +265,12 @@ async def _validate_proposed_state(
         self,
         db: OrionDBInterface,
     ):
-        if self.proposed_state is not None:
+        if (
+            self.response_status == SetStateStatus.REJECT
+            and self.proposed_state is None
+        ):
+            validated_orm_state = self.run.state
```

Comment:
Could we always do this if the proposed state is null? It seems nice to have it not just apply to REJECT semantics. Or do we perform logic based on `self.validated_state is None` later?

Reply:
I think we can! If our state machine acts like a markov chain, if orchestration doesn't specify a state only the current state makes sense as information